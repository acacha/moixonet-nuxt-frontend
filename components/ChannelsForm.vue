<template>
  <app-channels-form @added="add" />
</template>

<script>
import * as actionTypes from '../action-types'
import AppChannelsForm from './AppChannelsForm'

export default {
  name: 'ChannelsForm',
  components: {
    'app-channels-form': AppChannelsForm
  },
  methods: {
    add (channel) {
      this.$store.dispatch('channels/' + actionTypes.CHANNEL_STORE, channel)
    }
  }
}
</script>
