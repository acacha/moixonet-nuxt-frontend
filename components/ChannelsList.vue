<template>
  <v-container>
    <v-card>
      <v-toolbar dense color="primary">
        <v-toolbar-title class="white--text">Llista de canals</v-toolbar-title>
        <v-spacer></v-spacer>

        <v-toolbar-items>
          <v-btn text icon class="white--text" @click="refresh" :loading="loading">
            <v-icon>mdi-cached</v-icon>
          </v-btn>
        </v-toolbar-items>

      </v-toolbar>
      <v-list>
        <v-list-item v-for="channel in channels">
          {{ channel.name }}
        </v-list-item>
        <v-card-actions>
          <v-btn @click="omplir" color="primary">
            Omplir
          </v-btn>
          <v-btn @click="buidar" color="error">
            Buidar
          </v-btn>
          <v-btn @click="refresh">
            refresh
          </v-btn>
        </v-card-actions>
      </v-list>
    </v-card>
  </v-container>
</template>

<script>
// import axios from 'axios'
import channelsFixture from '../cypress/fixtures/channels'
export default {
  name: 'ChannelsList',
  data () {
    return {
      loading: false
    }
  },
  computed: {
    channels () {
      // return []
      return this.$store.getters['channels/channels']
    }
  },
  created () {
    this.refresh()
  },
  methods: {
    omplir () {
      // MODIFICAR L'ESTAT
      this.$store.commit('channels/SET_CHANNELS', channelsFixture)
    },
    buidar () {
      // MODIFICAR L'ESTAT
      this.$store.commit('channels/SET_CHANNELS', [])
    },
    async refresh () {
      this.loading = true
      await this.$store.dispatch('channels/REFRESH')
      this.loading = false
    },
    async refreshold () {
      try {
        const channels = await this.$axios.get('http://moixonet-backend.test/api/v1/channels')
        this.$store.commit('channels/SET_CHANNELS', channels.data)
      } catch (error) {
        console.log(error)
      }
    },
    refreshPromise () {
      this.$axios.get('http://moixonet-backend.test/api/v1/channels').then((channels) => {
        this.$store.commit('channels/SET_CHANNELS', channels.data)
      }).catch((error) => {
        console.log(error)
      })
    }
  }
}
</script>
