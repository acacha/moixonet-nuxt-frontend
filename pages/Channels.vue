<template>
  <span>
    <app-channels-form @added="add" />
    <app-channels-list :channels="channels" @removed="remove" />
  </span>
</template>

<script>
import AppChannelsList from '../components/AppChannelsList'
import AppChannelsForm from '../components/AppChannelsForm'
import channelsFixture from '../cypress/fixtures/channels'
export default {
  name: 'Channels',
  components: {
    'app-channels-list': AppChannelsList,
    'app-channels-form': AppChannelsForm
  },
  data () {
    return {
      channels: channelsFixture
    }
  },
  methods: {
    // operaciÃ³ sobre un element d'estat -> commit a Vuex
    add (name) {
      this.channels.push({
        id: 5,
        name
      })
    },
    remove (channel) {
      this.channels.splice(this.channels.indexOf(channel), 1)
    }
  }
}
</script>
