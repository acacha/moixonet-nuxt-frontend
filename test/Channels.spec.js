import { shallowMount } from '@vue/test-utils'
import Channels from '../pages/Channels'
import ChannelsList from '../components/ChannelsList'
import ChannelsForm from '../components/ChannelsForm'

describe('Channels', () => {
  let opts

  const build = (options) => {
    options = options || opts
    const wrapperShallow = shallowMount(Channels, options)

    return {
      wrapperShallow,
      channelsList: () => wrapperShallow.find(ChannelsList),
      channelsForm: () => wrapperShallow.find(ChannelsForm)
    }
  }

  beforeEach(() => {
    opts = {}
  })

  test('is a Vue instance', () => {
    const { wrapperShallow } = build()
    expect(wrapperShallow.isVueInstance()).toBeTruthy()
  })

  test('renders the component', () => {
    const { wrapperShallow } = build()
    expect(wrapperShallow.element).toMatchSnapshot()
  })

  test('renders child components', () => {
    const { channelsList, channelsForm } = build()
    expect(channelsList().exists()).toBe(true)
    expect(channelsForm().exists()).toBe(true)
  })
})
