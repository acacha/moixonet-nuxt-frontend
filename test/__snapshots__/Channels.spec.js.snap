// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channels renders the component 1`] = `
<span
  channels="[object Object],[object Object],[object Object]"
>
  <app-channels-form-stub />
   
  <app-channels-list-stub
    channels="[object Object],[object Object],[object Object]"
  />
</span>
`;
